### PR TITLE
Fix l10n gen file's problems

### PR DESCRIPTION
**Pull request by shirne**
Fix l10n generated file's no_leading_underscores_for_local_identifiers and require_trailing_commas problem

#137313

- [x] fix supported locale list `require_trailing_commas` in app_localizations.dart
- [x] add ignore `no_leading_underscores_for_local_identifiers` to localizations sub classes when has local_identifiers
- [x] fix `require_trailing_commas` for NumberFormat.currency code
- [x] fix `use_super_parameters` in localizations sub classes

Problems that still exist `unnecessary_string_interpolations` and `unnecessary_string_escapes` in localizations sub classes

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by eliasyishak** [comment]
@shirne thank you for contribution! Per the bot that commented, can you provide a test to confirm your changes are working as expected?

**Comment by shirne** [comment]
> @shirne thank you for contribution! Per the bot that commented, can you provide a test to confirm your changes are working as expected?

OK, I'll try it

**Comment by shirne** [comment]
@eliasyishak 
I added a test to the original gen_l10n.
Could you please take a look and see if there is anything else that needs to be done?

File: packages/flutter_tools/test/integration.shard/gen_l10n_test.dart [hunk 1]
```diff
@@ -180,4 +182,52 @@ void main() {
...
   });
+
+  // Analyze generated l10n file
+  // (see https://github.com/flutter/flutter/issues/137313)
+  testUsingContext('generated l10n file without problems', () async {
+    if (!fileSystem.file('${tempDir.path}/pubspec.yaml').existsSync()) {
+      await project.setUpIn(tempDir);
+    }
+    await globals.processManager.run(
+      <String>['flutter', 'gen-l10n'],
```
Comment (christopherfujino) [inline packages/flutter_tools/test/integration.shard/gen_l10n_test.dart:193]: i think expecting `flutter` to be on the path is the wrong thing to do here. Please use the FlutterTestDriver like the other tests in this file.

**Comment by christopherfujino** [comment]
Please fix the broken tests

**Comment by christopherfujino** [comment]
Closing this PR as stale as there hasn't been an update in 3 weeks. If you are interested in picking it back up, feel free to re-open it and address review feedback. Thanks for contributing!

**Comment by shirne** [comment]
> Closing this PR as stale as there hasn't been an update in 3 weeks. If you are interested in picking it back up, feel free to re-open it and address review feedback. Thanks for contributing!

I can't find a way to use FlutterTestDriver to determine lints problems

**Pull request closed without merging**